packages/service-clients/end-to-end-tests/azure-client/src/test/gc.spec.ts

```diff
@@ -0,0 +1,142 @@
+			if (isEphemeral) {
+				const containerResponse: AxiosResponse | undefined = await createContainerFromPayload(
+					ephemeralSummaryTrees.createContainerWithSharedTree,
+					"test-user-id-1",
+					"test-user-name-1",
+				);
+				const containerId = getContainerIdFromPayloadResponse(containerResponse);
+				const result = await client.getContainer(containerId, schema, compatibilityMode);
+				return result.container;
+			}
+
+			const { container } = await client.createContainer(schema, compatibilityMode);
+			await container.attach();
+			return container;
+		}
+
+		async function waitForConnected(container: IFluidContainer): Promise<void> {
+			if (container.connectionState !== ConnectionState.Connected) {
+				await timeoutPromise((resolve) => container.once("connected", () => resolve()), {
+					durationMs: connectTimeoutMs,
+					errorMsg: "container connect() timeout",
+				});
+			}
+		}
+
+		/**
+		 * Validates that the GarbageCollector event is logged and the GC configs are as expected.
+		 * @param compatibilityMode - Used in assert messages to tell which compat mode the validation is running in.
+		 */
+		function validateGCConfigs(compatibilityMode: CompatibilityMode): void {
+			const gcConfigString = mockLogger.events.find(
+				(event) =>
+					event.eventName ===
+					"fluid:telemetry:ContainerRuntime:GarbageCollector:GarbageCollectorLoaded",
+			)?.gcConfigs as string;
+			assert(
+				gcConfigString !== undefined,
+				`GC configs not found for compat mode ${compatibilityMode}`,
+			);
+			const gcConfigs = JSON.parse(gcConfigString) as IGarbageCollectorConfigs;
+			// Sweep should be enabled for documents but it should not run. Tombstone features should also be disabled.
+			assert.strictEqual(
+				gcConfigs.gcEnabled,
+				true,
+				`GC not enabled for compat mode ${compatibilityMode}`,
+			);
+			assert.strictEqual(
+				gcConfigs.sweepEnabled,
+				true,
+				`Sweep not enabled for compat mode ${compatibilityMode}`,
+			);
+			assert.strictEqual(
+				gcConfigs.shouldRunSweep,
+				"NO",
+				`Sweep should not run for compat mode ${compatibilityMode}`,
+			);
+			assert.strictEqual(
+				gcConfigs.throwOnTombstoneLoad,
+				false,
+				`Tombstone should not throw on load for compat mode ${compatibilityMode}`,
+			);
+			assert.strictEqual(
+				gcConfigs.throwOnTombstoneUsage,
+				false,
+				`Tombstone should not throw on usage for compat mode ${compatibilityMode}`,
+			);
+			assert.strictEqual(
+				gcConfigs.tombstoneAutorecoveryEnabled,
+				false,
+				`Tombstone autorecovery should be disabled for compat mode ${compatibilityMode}`,
+			);
+		}
+
+		/**
+		 * Garbage collection should be enabled along with sweep for documents created via AzureClient. However, sweep
+		 * along with tombstone should be disabled. Basically, these documents are eligible for sweep but sweep won't
+		 * run on them unless explicitly enabled via configs.
+		 */
+		it("should have sweep and tombstone disabled", async () => {
```

Comment:
@markfields made me realize that this test is basically the same as the one here - https://github.com/microsoft/FluidFramework/blob/3c6bfc3d429285b568bdfae417accfcaa5e0e190/packages/service-clients/azure-client/src/test/AzureClient.spec.ts#L289.

I will update this to do more than just checking for configs.

Reply:
Sounds good.  FWIW I prefer this approach rather than the casting and unsafe member access.